apps/beeswax/src/beeswax/server/dbms.py

```diff
@@ -143,14 +148,19 @@ def get_query_server_config(name='beeswax', connector=None):
           LOG.info("Setting up Hive with the following node {0}".format(znode))
           if zk.exists(znode):
             hiveservers = zk.get_children(znode)
-            server_to_use = 0 # if CONF.HIVE_SPREAD.get() randint(0, len(hiveservers)-1) else 0
-            cache.set("hiveserver2", json.dumps({"host": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[0], "port": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[1]}))
+            server_to_use = abs(hash(str(username))) % 10 % len(hiveservers)
+            if hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[0] == DBMS_OLDHS2_NAME_CACHE.get(username):
+              server_to_use = (server_to_use + 1) % len(hiveservers)
+            DBMS_OLDHS2_NAME_CACHE[username] = hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[0]
+            cache_user_host_port[username] = json.dumps({"host": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[0], "port": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[1]})
+            cache.set("hiveserver2", cache_user_host_port)
```

Comment:
I'm concerned that we're saving information for all users on each call. Would it make sense to only save the server for that user by using a key for each user e.g.:
`cache.set("hiveserver2%s" % user.id, json.dumps({"host": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[0], "port": hiveservers[server_to_use].split(";")[0].split("=")[1].split(":")[1]}))`

Reply:
My purpose is to allocate different hiveserver2 servers for different users to reduce the load capacity of one hiveserver2

Reply:
At the time I was not sure if django's cache could set multiple key-value pairs